### PR TITLE
[DOC] Update segment replication to include alerting plugin limitation

### PR DESCRIPTION
**Pull request by vagimeli**
### Description
Update segment replication to include alerting plugin limitation

### Issues Resolved
https://github.com/opensearch-project/documentation-website/issues/4967

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@dreamer-89 Please provide your technical review of the updated text in lines 25-27. Thank you, Melissa 

cc: @Naarcha-AWS

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 1]
```diff
@@ -51,12 +56,12 @@ PUT /my-index1
 In segment replication, the primary shard is usually generating more network traffic than the replicas because it copies segment files to the replicas. Thus, it's beneficial to distribute primary shards equally between the nodes. To ensure balanced primary shard distribution, set the dynamic `cluster.routing.allocation.balance.prefer_primary` setting to `true`. For more information, see [Cluster settings]({{site.url}}{{site.baseurl}}/api-reference/cluster-api/cluster-settings/).
 
 Segment replication currently does not support the `wait_for` value in the `refresh` query parameter.
```
Comment (dreamer-89) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:58]: Do we need to keep all limitations at one place under limitation sections ?

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 2]
```diff
@@ -22,19 +22,24 @@ When the primary shard sends a checkpoint to replica shards on a refresh, a new
 
 Segment replication is the first feature in a series of features designed to decouple reads and writes in order to lower compute costs.
 
+**Limitation:** Segment replication is not compatible with the [Alerting]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/) plugin. The Alerting plugin relies on the latest data being available on replica shards when using the `immediate` refresh policy. Segment replication does not provide the same guarantee, which means shards can become stale. This especially impacts [document-level monitors]({{site.url}}{{site.baseurl}}/observing-your-data/alerting/api/#document-level-monitors).
```
Comment (dreamer-89) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:25]: >Segment replication does not provide the same guarantee, which means shards can become stale.

It is not clear what do you mean by same guarantee. Can you elaborate more on this ?

Comment (dreamer-89) [reply _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:25]: @lezzago @anasalkouz Can you also review this content.

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md (hunk 1 above)
Comment (vagimeli) [reply _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:58]: Yes, good catch. Change made.

Comment (vagimeli) [reply _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:58]: Do we have documentation for the refresh parameter?

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md (hunk 2 above)
Comment (vagimeli) [reply _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:25]: @dreamer-89 Revised. Please review for accuracy.

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 3]
```diff
@@ -22,19 +22,24 @@ When the primary shard sends a checkpoint to replica shards on a refresh, a new
 
 Segment replication is the first feature in a series of features designed to decouple reads and writes in order to lower compute costs.
 
+**Limitation:** Segment replication is not compatible with the [Alerting]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/) plugin. The Alerting plugin relies on the latest data being available on replica shards when using the `immediate` refresh policy. The `immediate` refresh policy in segment replication can resource-intensive and introduce latency due to real-time data synchronization. As such, segment replication cannot guarantee the availability of the `immediate` refresh policy and thus lead to stale replica shards. This especially impacts [document-level monitors]({{site.url}}{{site.baseurl}}/observing-your-data/alerting/api/#document-level-monitors). Additionally, segment replication does not support the `wait_for` value in the `refresh` query parameter.
```
Comment (anasalkouz) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:25]: Segment replication is not compatible with the [Alerting] is very broad term sentence. I think we need to be more specific. For example Segment replication is not compatible with the document-level monitors in Alerting Plugin.

Can you add that this is a temporary limitation? and we are going to fix this limitation in future releases by routing queries to primary shard

Comment (lezzago) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:25]: We also need to call out that Security Analytics is not compatible at all with Segment replication since it heavily uses Document Level monitors behind the scenes.

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 4]
```diff
@@ -106,11 +100,13 @@ When using segment replication, consider the following:
...
+1. Read-after-write guarantees: The `wait_until` and `wait_for` refresh policy is not compatible with segment replication. If you use the `wait_until` refresh policy while ingesting documents, you'll get a response only after the primary node has refreshed and made those documents searchable. Replica shards will respond only after having written to their local translog. We are exploring other mechanisms for providing read-after-write guarantees. For more information, see the corresponding [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/6046).  
 1. System indexes will continue to use document replication internally until read-after-write guarantees are available. In this case, document replication does not hinder the overall performance because there are few system indexes.
-
+1. Segment replication is not compatible with [security analytics]({{site.url}}{{site.baseurl}}/security-analytics/index/). Security analytics relies on being able to access the latest data in an index, and segment replication can delay the availability of this data.
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:108]: Can we move this point to right after document-level monitors and say 
Segment replication is not compatible with security analytics because security analytics uses document-level monitors?

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 5]
```diff
@@ -106,11 +100,13 @@ When using segment replication, consider the following:
...
-1. Read-after-write guarantees: The `wait_until` refresh policy is not compatible with segment replication. If you use the `wait_until` refresh policy while ingesting documents, you'll get a response only after the primary node has refreshed and made those documents searchable. Replica shards will respond only after having written to their local translog. We are exploring other mechanisms for providing read-after-write guarantees. For more information, see the corresponding [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/6046).  
+1. Read-after-write guarantees: The `wait_until` and `wait_for` refresh policy is not compatible with segment replication. If you use the `wait_until` refresh policy while ingesting documents, you'll get a response only after the primary node has refreshed and made those documents searchable. Replica shards will respond only after having written to their local translog. We are exploring other mechanisms for providing read-after-write guarantees. For more information, see the corresponding [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/6046).
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:106]: Actually, the policy is internally has an enum value `WAIT_UNTIL` but the value set in a request is `wait_for`. So they are the same thing: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/support/WriteRequest.java#L97
```suggestion
1. Read-after-write guarantees: Segment replication currently does not support setting refresh policy to `wait_for`.  If you set the `refresh` query parameter to `wait_for` and then ingest documents, you'll get a response only after the primary node has refreshed and made those documents searchable. Replica shards will respond only after having written to their local translog. We are exploring other mechanisms for providing read-after-write guarantees. For more information, see the corresponding [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/6046).  
```

**Review by kolchfa-aws — COMMENTED** [review]
A couple of comments and please fix links. Thanks!

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md (hunk 3 above)
Comment (vagimeli) [reply _tuning-your-cluster/availability-and-recovery/segment-replication/index.md]: Revised

Comment (vagimeli) [reply _tuning-your-cluster/availability-and-recovery/segment-replication/index.md]: Revised

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md (hunk 2 above)
Comment (vagimeli) [reply _tuning-your-cluster/availability-and-recovery/segment-replication/index.md]: Revised

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md (hunk 1 above)
Comment (vagimeli) [reply _tuning-your-cluster/availability-and-recovery/segment-replication/index.md]: Revised

**Review by kolchfa-aws — APPROVED** [review]
LGTM

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 6]
```diff
@@ -106,11 +100,13 @@ When using segment replication, consider the following:
 
 1. Enabling segment replication for an existing index requires [reindexing](https://github.com/opensearch-project/OpenSearch/issues/3685).
 1. [Cross-cluster replication](https://github.com/opensearch-project/OpenSearch/issues/4090) does not currently use segment replication to copy between clusters.
+1. Segment replication is not compatible with [document-level monitors]({{site.url}}{{site.baseurl}}/observing-your-data/alerting/api/#document-level-monitors), which are used with the [Alerting]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/) plugin. The Alerting plugin uses the latest available data on replica shards when using the `immediate` refresh policy, and segment replication can delay the policy's availability, resulting in stale replica shards.
+1. Segment replication is not compatible with [security analytics]({{site.url}}{{site.baseurl}}/security-analytics/index/). Security analytics uses document-level monitors, which are not compatible with segment replication.
```
Comment (anasalkouz) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:104]: Is this duplicate?

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 7]
```diff
@@ -12,29 +12,29 @@ redirect_from:
 
 # Segment replication
 
-With segment replication, segment files are copied across shards instead of documents being indexed on each shard copy. This improves indexing throughput and lowers resource utilization at the expense of increased network utilization.
+Segment replication involves copying segment files across shards instead of indexing documents on each shard copy. This approach enhances indexing throughput and reduces resource utilization, but increases network utilization. Segment replication is the first feature in a series of features designed to decouple reads and writes in order to lower compute costs.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:15]:
```suggestion
Segment replication involves copying segment files across shards instead of indexing documents on each shard copy. This approach enhances indexing throughput and reduces resource utilization but increases network utilization. Segment replication is the first feature in a series of features designed to decouple reads and writes in order to lower compute costs.
```

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 8]
```diff
@@ -12,29 +12,29 @@ redirect_from:
...
 - When there are segment file changes on a primary shard refresh.
 - During peer recovery, such as replica shard recovery and shard relocation (explicit allocation using the `move` allocation command or automatic shard rebalancing).
 
-Segment replication is the first feature in a series of features designed to decouple reads and writes in order to lower compute costs.
-
 ## Use cases
 
+Segment replication can be applied in a variety of scenarios, including:
+
 - Users who have high write loads but do not have high search requirements and are comfortable with longer refresh times.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:27]: "Users" are not "scenarios", so the first two items do not complete the stem. The stem is fine, but I would rephrase the first two items.

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 9]
```diff
@@ -78,15 +75,12 @@ cluster.indices.replication.strategy: 'SEGMENT'
 ```
 {% include copy.html %}
 
-This cluster-level setting cannot be enabled through the REST API.
-{: .note}
-
-This setting is not applied to system indexes and hidden indexes. By default, all system and hidden indexes in OpenSearch will still use document replication even if this setting is enabled.
+This cluster-level setting cannot be enabled through the [REST API]({{site.url}}{{site.baseurl}}/api-reference/index/). This setting is not applied to system indexes and hidden indexes. By default, all system and hidden indexes in OpenSearch use document replication even if this setting is enabled.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:78]:
```suggestion
This cluster-level setting cannot be enabled through the [REST API]({{site.url}}{{site.baseurl}}/api-reference/index/). This setting is not applied to system indexes and hidden indexes. By default, all system and hidden indexes in OpenSearch use document replication, even if this setting is enabled.
```

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md (hunk 6 above)
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:104]: Should both instances of "Security Analytics" be capitalized as a proper noun?

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 10]
```diff
@@ -106,11 +100,13 @@ When using segment replication, consider the following:
...
-1. Read-after-write guarantees: The `wait_until` refresh policy is not compatible with segment replication. If you use the `wait_until` refresh policy while ingesting documents, you'll get a response only after the primary node has refreshed and made those documents searchable. Replica shards will respond only after having written to their local translog. We are exploring other mechanisms for providing read-after-write guarantees. For more information, see the corresponding [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/6046).  
+1. Read-after-write guarantees: Segment replication currently does not support setting refresh policy to `wait_for`.  If you set the `refresh` query parameter to `wait_for` and then ingest documents, you'll get a response only after the primary node has refreshed and made those documents searchable. Replica shards will respond only after having written to their local translog. We are exploring other mechanisms for providing read-after-write guarantees. For more information, see the corresponding [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/6046).
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:107]:
```suggestion
1. Read-after-write guarantees: Segment replication does not currently support setting the refresh policy to `wait_for`.  If you set the `refresh` query parameter to `wait_for` and then ingest documents, you'll get a response only after the primary node has refreshed and made those documents searchable. Replica shards will respond only after having written to their local translog. We are exploring other mechanisms for providing read-after-write guarantees. For more information, see the corresponding [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/6046).  
```

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 11]
```diff
@@ -398,4 +390,4 @@ The benchmarking results show a non-zero error rate as the number of replicas in
 ## Next steps
 
 1. Track [future enhancements to segment replication](https://github.com/orgs/opensearch-project/projects/99).
-1. Read [this blog post about segment replication](https://opensearch.org/blog/segment-replication/).
\ No newline at end of file
+1. Read this [blog post about segment replication](https://opensearch.org/blog/segment-replication/).
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:393]:
```suggestion
1. Read [this blog post about segment replication](https://opensearch.org/blog/segment-replication/).
```

**Review by natebower — COMMENTED** [review]
@vagimeli Please see my comments and changes and let me know if you have any questions. Thanks!

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md (hunk 6 above)
Comment (vagimeli) [reply _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:104]: Revised to combine into one bullet point.

Comment (vagimeli) [reply _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:104]: Yes, as we're referring to the Security Analytics plugin.

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md (hunk 8 above)
Comment (vagimeli) [reply _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:27]: Revised.

**Comment by vagimeli** [comment]
@natebower Please review the changes made to address comments in lines 27-28 and 103-104. Thanks!

**Review by natebower — APPROVED** [review]
LGTM!
